src/components/categories-filter/CategoriesFilter.component.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useState } from 'react';
+
+import appStyles from '@/App.module.css';
+import { CategoriesDataProviderComponent } from '@/data-providers/CategoriesDataProvider.component.tsx';
+import { transformCheckedCategories, transformCheckedCategoryIds } from '@/helpers/categoriesFilterHelper.ts';
+import type { CheckedCategoriesProps } from '@/types/CheckedCategoriesProps.type.tsx';
+
+import styles from './categories-filter.module.css';
+
+interface CategoriesFilterProps {
+    categoryIds?: number[];
+    onCategoriesChange: (categoryIds: number[]) => void;
+}
+
+export const CategoriesFilterComponent: React.FC<CategoriesFilterProps> = ({ categoryIds, onCategoriesChange }) => {
+    const [checkedCategories, setCheckedCategories] = useState<CheckedCategoriesProps>(() => transformCheckedCategories(categoryIds || []));
+
+    const handleCategoryCheck = (event: React.ChangeEvent<HTMLInputElement>) => {
+        const target = event.target as HTMLInputElement;
+        const updatedCheckedCategories = { ...checkedCategories, [target.value]: target.checked };
+
+        setCheckedCategories(updatedCheckedCategories);
+        onCategoriesChange(transformCheckedCategoryIds(updatedCheckedCategories));
+    };
+
+    return (
+        <div className={styles.categoryFilterWrapper}>
+            <CategoriesDataProviderComponent>
+                {({ categories }) => (
+                    <>
+                        {categories.map((categoryData) => (
+                            <div key={categoryData.id}>
+                                <input
```

Comment:
Would be good to add a bit of semantic
```suggestion
                    <ul>
                        {categories.map((categoryData) => (
                            <li key={categoryData.id}>
                                <input
```

Reply:
fixed